modules/genomic_browser/php/NDB_Menu_Filter_cpg_browser.class.inc

```diff
@@ -0,0 +1,429 @@
+        $this->addBasicText('PSCID', 'PSCID:');
+        $this->addBasicText('External_ID', 'External ID:');
+        $this->addSelect(
+            'gender',
+            'Gender:',
+            array(
+             ''       => 'All',
+             'Male'   => 'Male',
+             'Female' => 'Female',
+            )
+        );
+        $this->addSelect(
+            'SubprojectID',
+            'Subproject:',
+            array('' => 'Any') + $list_of_subprojects
+        );
+        $this->addBasicText('dob', 'Date of Birth:', array('maxlength' => 10));
+
+        $show_results_options
+            = array(
+               'brief' => 'Summary fields',
+               'full'  => 'All fields',
+              );
+        $this->addSelect('Show_Brief_Results', 'Display:', $show_results_options);
+
+        // CpG
+        $this->addBasicText('CpG', 'CpG:');
+        $Base_options = array(
+                         null      => 'Any',
+                         'Island'  => 'Island',
+                         'N_Shore' => 'North shore',
+                         'S_Shelf' => 'South shelf',
+                         'S_Shore' => 'South shore',
+                        );
+        $this->addSelect('Context', 'Context:', $Base_options);
+        $this->addBasicText('Date_collected', 'Date collected:');
+        $Base_options = array(
+                         null               => 'Any',
+                         'High-methylation' => 'High-methylation',
+                         'Partially'        => 'Partially',
+                         'Low-methylation'  => 'Low-methylation',
+                        );
+        $this->addSelect('Methylation', 'Methylation:', $Base_options);
+        $this->addBasicText('Beta_value', 'Beta-value:');
+        $Base_options = array(
+                         null     => 'Any',
+                         'Buccal' => 'Buccal',
+                        );
+        $this->addSelect('Tissue', 'Tissue:', $Base_options);
+
+        // Gene
+        $this->addBasicText('Chromosome', 'Chromosome:');
+        $Base_options = array(
+                         null     => 'Any',
+                         'GRCh37' => 'GRCh37',
+                        );
+        $this->addSelect('Assembly', 'Build:', $Base_options);
+        $Base_options = array(
+                         null => 'Any',
+                         'F'  => 'Foward',
+                         'R'  => 'Reverse',
+                        );
+        $this->addSelect('Strand', 'Strand:', $Base_options);
+        $this->addBasicText('Gene', 'Symbol:');
+
+        $DB = Database::singleton();
+        $platform_results
+            = $DB->select("SELECT distinct platform FROM genomic_cpg ");
+        $platform_options = array('' => 'Any');
+        foreach ($platform_results as $row => $result ) {
+            $name = $result['platform'];
+            $platform_options[$name] = $name;
+        }
+        $this->addSelect('Platform', 'Platform:', $platform_options);
```

Comment:
Platform would ideally be drawn from (a) Platform table.  Right now we only have genotyping_platform table.  Let's look into generalizing this table in future for use with (e.g.) CpG.